lib/services/local/generic/generic_test.go

```diff
@@ -52,7 +52,10 @@ func newTestResource(name string) *testResource {
 		},
 	}
 
-	tr.CheckAndSetDefaults()
+	err := tr.CheckAndSetDefaults()
+	if err != nil {
+		panic(err)
```

Comment:
Could we do better than a panic here?

Reply:
Yeah, this is only a test but I've added a nicer check.